### PR TITLE
Fix NUMA strategy cast in ResourceManager::move call

### PR DESCRIPTION
**Pull request by jdahm**

### CODE REVIEW

File: src/umpire/ResourceManager.cpp [hunk 1]
```diff
@@ -565,21 +565,8 @@ ResourceManager::reallocate(void* src_ptr, std::size_t size, Allocator allocator
 #if defined(UMPIRE_ENABLE_NUMA)
 static strategy::NumaPolicy* cast_as_numa_policy(Allocator& allocator) {
```
Comment (davidbeckingsale) [inline src/umpire/ResourceManager.cpp]: Why don't we just use `unwrap_allocator` directly (I'm unsure what this function is helping with?)

Comment (jdahm) [reply src/umpire/ResourceManager.cpp]: `unwrap_allocator` throws an error if it can't cast as that type

**Comment by jdahm** [comment]
@davidbeckingsale is this better?

File: CHANGELOG.md [hunk 2]
```diff
@@ -20,15 +20,19 @@ Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to
 
 ### Changed
 
-- Adjust notifications for CI jobs
-- Use git commit hash as RC version in develop builds
-- Update BLT submodule to fix warnings from CMake 3.14 and warnings from HIP library
+- Adjust notifications for CI jobs.
+
```
Comment (davidbeckingsale) [inline CHANGELOG.md]: Why adding blank lines?

Comment (jdahm) [reply CHANGELOG.md]: For consistency.

**Comment by jdahm** [comment]
I'm going to double check that this works on LC, then merge.

**Pull request merged**